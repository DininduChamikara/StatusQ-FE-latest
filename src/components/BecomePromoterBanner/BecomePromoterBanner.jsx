import { Poll, WorkHistory } from "@mui/icons-material";
import { Box, Button, Paper, Stack, Typography } from "@mui/material";
import React from "react";
import PromotersImage from "../../images/promoter.jpg";

function BecomePromoterBanner({ setVisibleStepper }) {
  return (
    <Paper
      data-testid="becomePromoterBanner"
      elevation={3}
      sx={{
        p: 3,
        width: "100%",
        display: { xs: "block", lg: "flex" },
        alignItems: "center",
        flexDirection: "row",
        justifyContent: "center",
      }}
    >
      <Box
        sx={{
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "center",
          width: { xs: "100%", lg: "50%" },
        }}
      >
        <img height={300} src={PromotersImage} alt={""} />
      </Box>
      <Box
        sx={{ width: { xs: "100%", lg: "50%" }, alignItems: "center", p: 2 }}
      >
        <Typography
          sx={{
            fontSize: "1.8rem",
            fontWeight: "bold",
            mt: 2,
            color: "primary.darker",
          }}
        >
          Become a promoter
        </Typography>
        <Typography
          sx={{
            fontSize: "1.8rem",
            fontWeight: "bold",
            mb: 2,
            color: "primary.darker",
          }}
        >
          & earn money
        </Typography>
        <Typography sx={{ color: "primary.dark" }}>
          If you want to become a promoter, first you need to complete the
          promoter survey...!
        </Typography>
        <Box sx={{ my: 2 }}>
          <Stack direction="row" spacing={2}>
            <Button
              data-testid="becomePromoterBanner-btn-1"
              sx={{ width: 180 }}
              variant="contained"
              color="primary"
              startIcon={<Poll />}
              onClick={() => {
                setVisibleStepper(true);
              }}
            >
              Take the Survay
            </Button>
            <Button
              data-testid="becomePromoterBanner-btn-2"
              sx={{ width: 180 }}
              variant="contained"
              color="secondary"
              startIcon={<WorkHistory />}
              onClick={() => {
                setVisibleStepper(false);
              }}
            >
              View Jobs
            </Button>
          </Stack>
        </Box>
      </Box>
    </Paper>
  );
}

export default BecomePromoterBanner;
