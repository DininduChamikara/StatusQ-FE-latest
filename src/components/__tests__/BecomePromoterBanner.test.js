import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import BecomePromoterBanner from "../BecomePromoterBanner/BecomePromoterBanner";

describe("BecomePromoterBanner", () => {
  it("should render BecomePromoterBanner", () => {
    render(<BecomePromoterBanner />);
    const becomePromoterBannerElement = screen.getByTestId(
      "becomePromoterBanner"
    );
    expect(becomePromoterBannerElement).toBeInTheDocument();
  });

  it("BecomePromoterBanner should visible", () => {
    render(<BecomePromoterBanner />);
    const becomePromoterBannerElement = screen.getByTestId(
      "becomePromoterBanner"
    );
    expect(becomePromoterBannerElement).toBeVisible();
  });

  it ("availability of survey button", () => {
    // eslint-disable-next-line no-unused-vars
    const {getByRole} = render(<BecomePromoterBanner />);
    const takeSurveyBtn = screen.getByRole("button", {name:"Take the Survay"});
    expect(takeSurveyBtn).toBeInTheDocument();
  })

  it ("visibility of survey button", () => {
    // eslint-disable-next-line no-unused-vars
    const {getByRole} = render(<BecomePromoterBanner />);
    const takeSurveyBtn = screen.getByRole("button", {name:"Take the Survay"});
    expect(takeSurveyBtn).toBeVisible();
  })

  it ("survey button should be clickable", () => {
    const mockSetVisibleStepper = jest.fn();
    render(<BecomePromoterBanner setVisibleStepper={mockSetVisibleStepper} />);
    const becomePromoterBannerBtn_1 = screen.getByTestId(
      "becomePromoterBanner-btn-1"
    );
    fireEvent.click(becomePromoterBannerBtn_1);  
    expect(mockSetVisibleStepper).toHaveBeenCalledTimes(1);
  })

  it ("view jobs button should be clickable", () => {
    const mockSetVisibleStepper = jest.fn();
    render(<BecomePromoterBanner setVisibleStepper={mockSetVisibleStepper} />);
    const becomePromoterBannerBtn_2 = screen.getByTestId(
      "becomePromoterBanner-btn-2"
    );
    fireEvent.click(becomePromoterBannerBtn_2);  
    expect(mockSetVisibleStepper).toHaveBeenCalledTimes(1);
  })

  it("BecomePromoterBanner matches snapshot", () => {
    const tree = renderer.create(<BecomePromoterBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
