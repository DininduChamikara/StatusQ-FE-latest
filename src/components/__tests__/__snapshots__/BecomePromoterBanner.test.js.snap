// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BecomePromoterBanner BecomePromoterBanner matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-2plwtp-MuiPaper-root"
  data-testid="becomePromoterBanner"
>
  <div
    className="MuiBox-root css-19ccq3k"
  >
    <img
      alt=""
      height={300}
      src="promoter.jpg"
    />
  </div>
  <div
    className="MuiBox-root css-14a5gor"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-clv7s8-MuiTypography-root"
    >
      Become a promoter
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1pohf7y-MuiTypography-root"
    >
      & earn money
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-gtcsjv-MuiTypography-root"
    >
      If you want to become a promoter, first you need to complete the promoter survey...!
    </p>
    <div
      className="MuiBox-root css-1ebprri"
    >
      <div
        className="css-e53awj-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-gx9rqm-MuiButtonBase-root-MuiButton-root"
          data-testid="becomePromoterBanner-btn-1"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PollIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
              />
            </svg>
          </span>
          Take the Survay
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ztba17-MuiButtonBase-root-MuiButton-root"
          data-testid="becomePromoterBanner-btn-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="WorkHistoryIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 11c1.49 0 2.87.47 4 1.26V8c0-1.11-.89-2-2-2h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h7.68c-.43-.91-.68-1.92-.68-3 0-3.87 3.13-7 7-7zm-8-7h4v2h-4V4z"
              />
              <path
                d="M18 13c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm1.65 7.35L17.5 18.2V15h1v2.79l1.85 1.85-.7.71z"
              />
            </svg>
          </span>
          View Jobs
        </button>
      </div>
    </div>
  </div>
</div>
`;
